javascript/ql/lib/semmle/javascript/frameworks/Hapi.qll

```diff
@@ -9,10 +9,31 @@ module Hapi {
   /**
    * An expression that creates a new Hapi server.
    */
-  class ServerDefinition extends Http::Servers::StandardServerDefinition, DataFlow::NewNode {
+  class ServerDefinition extends Http::Servers::StandardServerDefinition, DataFlow::Node {
     ServerDefinition() {
       // `server = new Hapi.Server()`
       this = DataFlow::moduleMember("hapi", "Server").getAnInstantiation()
+      or
+      // `server = Glue.compose(manifest, composeOptions)`
+      this = DataFlow::moduleMember("@hapi/glue", "compose").getAnInvocation()
+      or
+      // `register (server, options)`
+      // `module.exports.plugin = {register, pkg};`
+      this =
+        any(NodeModule m)
+            .getAnExportedValue("plugin")
+            .(DataFlow::ObjectLiteralNode)
+            .getAPropertySource("register")
+            .(DataFlow::FunctionNode)
+            .getParameter(0)
+      or
+      // `const after = function (server) {...};`
+      // `server.dependency('name', after);`
+      exists(ServerDefinition server, DataFlow::MethodCallNode call |
+        call = server.ref().getAMethodCall() and
+        call.getMethodName() = "dependency" and
+        this = call.getArgument(1).(DataFlow::FunctionNode).getParameter(0)
```

Comment:
We have a utility function that fits.  
That also handles edge-cases around e.g. bound parameters.  

```suggestion
        this = call.getABoundCallbackParameter(1, 0)
```